src/components/button.js

```diff
@@ -37,8 +37,18 @@ module.exports = function(app) {
             }
           };
 
+
+         var allowSubmission = true;
           $scope.hasError = function() {
-            return clicked && (settings.action === 'submit') && $scope.formioForm.$invalid && !$scope.formioForm.$pristine;
+            var errValue = clicked && (settings.action === 'submit') && $scope.formioForm.$invalid && !$scope.formioForm.$pristine;
+            if(errValue && settings.disableOnInvalid && $scope.formioForm.$invalid && !$scope.formioForm.$pristine){
+              allowSubmission = false;
+              $('#' +settings.key).addClass("bttn-disable");
+            }else{
+              allowSubmission = true;
+              $('#'+settings.key).removeClass("bttn-disable");
```

Comment:
Do not use jQuery. Use ```angular.element``` instead. Also, you shouldn't need to search the whole DOM here. You should be able to pass ```$element``` to the controller to have access to the actual element for this directive.